lib/devfs/include/devfs/device.h

```diff
@@ -95,42 +104,62 @@ struct devops {
  * Driver object
  */
 struct driver {
-	const char	*name;		/* name of device driver */
-	struct devops	*devops;	/* device operations */
-	size_t		devsz;		/* size of private data */
-	int		flags;		/* state of driver */
+	/* name of device driver */
+	const char *name;
+	/* device operations */
+	struct devops *devops;
+	/* size of private data */
+	size_t devsz;
+	/* state of driver */
+	int flags;
 };
 
 /*
  * flags for the driver.
  */
 
 typedef enum device_state {
-	DS_INACTIVE	    = 0x00,		/* driver is inactive */
-	DS_ALIVE	    = 0x01,		/* probe succeded */
-	DS_ACTIVE	    = 0x02,		/* intialized */
-	DS_DEBUG	    = 0x04,		/* debug */
-	DS_NOTPRESENT   = 0x08,     /* not probed or probe failed */
-	DS_ATTACHING    = 0x10,     /* currently attaching */
-	DS_ATTACHED     = 0x20,     /*attach method called */
+	/* Driver is inactive */
+	DS_INACTIVE	    = 0x00,
+	/* Probe succeded */
+	DS_ALIVE	    = 0x01,
+	/* Intialized */
+	DS_ACTIVE	    = 0x02,
+	/* Debug */
+	DS_DEBUG	    = 0x04,
+	/* Not probed or probe failed */
+	DS_NOTPRESENT   = 0x08,
+	/* Currently attaching */
+	DS_ATTACHING    = 0x10,
+	/* Attach method called */
+	DS_ATTACHED     = 0x20,
 } device_state_t;
 
 /*
  * Device object
  */
 struct device {
-	struct device	*next;		/* linkage on list of all devices */
-	struct driver	*driver;	/* pointer to the driver object */
-	char		name[MAXDEVNAME]; /* name of device */
-	int		flags;		/* D_* flags defined above */
-	int		active;		/* device has not been destroyed */
-	int		refcnt;		/* reference count */
-	off_t		size;		/* device size */
-	off_t		offset; /* 0 for the main drive, if we have a
-				 *  partition, this is the start address
-				 */
-	size_t		max_io_size;
-	void		*private_data;	/* private storage */
+	/* Linkage on list of all devices */
+	struct device *next;
+	/* Pointer to the driver object */
+	struct driver *driver;
+	/* Name of device */
+	char name[MAXDEVNAME];
+	/* D_* flags defined above */
+	int flags;
+	/* Device has not been destroyed */
```

Comment:
Maybe we can change this to status of the device? And mention that 1 means active and 0 inactive. What do you think?

Reply:
Yeah, it's more clear description.